### PR TITLE
Made security patches by authorizing user for comments, follows, likes, and photos

### PR DESCRIPTION
**Pull request by dgarcia13-99**
Added authorization to photos, comments, likes, and follow_requests to only allow owners to edit and delete their own comments and photos as well as view their own feed and discover pages.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
+  before_action :is_an_authorized_user, only: [:destroy, :create]
```
Comment (jessjno) [inline app/controllers/comments_controller.rb:3]: good job creating the private method to use a before_action

File: app/views/photos/_photo.html.erb [hunk 2]
```diff
@@ -7,13 +7,16 @@
     </h2>
 
     <div>
+    <%if current_user == photo.owner%>
```
Comment (jessjno) [inline app/views/photos/_photo.html.erb:10]: nice job adding this if statement so users can't edit other users photos

File: config/routes.rb
Comment (jessjno) [inline config/routes.rb:1]: good job limiting the routes!

**Review by jessjno — APPROVED** [review]
good job!

File: app/controllers/follow_requests_controller.rb [hunk 3]
```diff
@@ -67,4 +68,11 @@ def set_follow_request
     def follow_request_params
       params.require(:follow_request).permit(:recipient_id, :sender_id, :status)
     end
+
+    def ensure_current_user_is_owner
```
Comment (marywebby) [inline app/controllers/follow_requests_controller.rb:72]: Looks like its working for both recipient and sender, but could also create this same logic for the `@follow_request.recipient` to check for update

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -66,4 +59,11 @@ def set_like
     def like_params
       params.require(:like).permit(:fan_id, :photo_id)
     end
+
+    def ensure_current_user_is_owner
+      if current_user != @like.fan
```
Comment (marywebby) [inline app/controllers/likes_controller.rb:64]: good implementation here

File: app/controllers/photos_controller.rb [hunk 5]
```diff
@@ -63,6 +64,12 @@ def set_photo
       @photo = Photo.find(params[:id])
     end
 
+    def ensure_current_user_is_owner
+      if current_user != @photo.owner
```
Comment (marywebby) [inline app/controllers/photos_controller.rb:68]: good job changing this!

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action :ensure_current_user_is_owner, only: [:feed, :discover]
```
Comment (marywebby) [inline app/controllers/users_controller.rb:3]: haven't seen people do the UsersController yet but i am liking this implementation here!

File: app/views/comments/_comment.html.erb [hunk 7]
```diff
@@ -9,14 +9,19 @@
       </h5>
       <p><%= comment.body %></p>
     </div>
+    
     <div>
+      <% if current_user == comment.author %>
```
Comment (marywebby) [inline app/views/comments/_comment.html.erb:14]: 👍

File: app/views/users/show.html.erb [hunk 8]
```diff
@@ -4,6 +4,7 @@
   </div>
 </div>
 
+<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
```
Comment (marywebby) [inline app/views/users/show.html.erb:7]: i liked what you did for liked and show!

File: config/routes.rb [hunk 9]
```diff
@@ -4,14 +4,14 @@
   devise_for :users
   
   resources :comments
```
Comment (marywebby) [inline config/routes.rb:6]: could add `except: [{index, :show]` here

**Review by marywebby — APPROVED** [review]
Great job on this Dani!!! I liked the way you refactored the users controller too!

**Pull request merged**